source/init/callback.cc

```diff
@@ -0,0 +1,80 @@
+#include "init/callback.h"
```

Comment:
This should probably be `source/common/init`, this isn't a top-level Envoy concern.

Reply:
I was wondering about that. The current state of `master` is sort of weird:

* The `Init::Manager` and `Init::Target` interfaces are defined in `include/envoy/init`
* The `InitManagerImpl` class is defined in `source/server` with a TODO comment, "Move InitManagerImpl into a new subdirectory in source/ called init/"

I agree it'd make sense to move the implementation into `source/common/init`. What would you think about also moving the interfaces down into `include/envoy/common/init`?

Reply:
`source/common/init` for implementation, `include/envoy/init` for interface; that's how Envoy canonical paths work.